tests/unit/development/test_base_component.py

```diff
@@ -55,71 +57,67 @@ def nested_tree():
     return c, c1, c2, c3, c4, c5
 
 
-class TestComponent(unittest.TestCase):
+class TestComponent:
```

Comment:
This is purely personal preferences, there is no right answer here. The purpose of keeping a class as a logical container is redundant in the sense that the test file name is doing the same thing, especially under the context of `pytest`.  another bonus point is that we can avoid `self`.  another practice is to use fixtures defined in `conftest.py` to share some common patterns in the test, which used to the reason for using `class` in `unittest`.

Reply:
+1 rather than classes, let's just break into more files, and organize that way.

Reply:
Sure, I would prefer as well